include/heap.h

```diff
@@ -42,7 +42,7 @@ extern "C" {
 #define HP_MAX_KEY                  MAX_INDEXES         /* Max allowed keys */
 #endif
 
-#define HP_MAX_KEY_LENGTH           1000            /* Max length in bytes */
+#define HP_MAX_KEY_LENGTH           MAX_KEY_LENGTH      /* Max length in bytes */
```

Comment:
I'm sorry for not commenting this earlier, but this is a functionality change for MEMORY SE, thus needs at least the most basic test: CREATE TABLE w/ index lengths > 1000 < new maximum, check insert, reads using that index, other reads. Check that CREATE TABLE w/ index lengths > newmax fails OK. Check that composite key with key parts > 1000 < newmax, sum < newmax work, sum > newmax fail.

Reply:
Uh, we have another issue there, and I was able to reproduce this in upstream, both 5.7 and 8.0:

varchar and varbinary ordering fails in memory tables if the key is longer than 1021 characters. (as the null terminator and the length adds 3 more bytes, as would be my guess)

As this is present in the upstream too, and it's a different bug than the crash, shouldn't we handle it as a different issue?

Reply:
I figured this out, and added a test (with a single key for now). The issue is that the max_sort_length variable defaults to 1024, while the max key length is 3072.

For now, I changed the max_sort_length value for the test sessions, but maybe it would be a good idea to change its default to be 3072 too - otherwise a query's result will depend on the sorting algorithm used.